.github/workflows/CI_Build.yml

```diff
@@ -32,5 +32,6 @@ jobs:
     - name: test lib on ${{ matrix.os }}
       run: |
         cd build
+        cmake .. -DCMAKE_BUILD_TYPE=Debug -G "Unix Makefiles"        
```

Comment:
Debug mode can be extremely slow for this code due to the absence of optimizations... (sometimes 10x or 100x slower) are you sure you want Debug here and not Release?

Reply:
When the Release mode is used those `assert()` statements are just passed without checking. 

In the Release mode, there must be a way to throw an error in C, but I need to learn how to do it.

Reply:
OK, you can leave Debug if it's just for CI tests... in any case you are considering extremely small instances so it should run. Additionally, I would suggest running for a fixed number of iterations in that CI test (e.g., 10000) using the available option. This would avoid extremely long traces (currently, the code runs until a time limit which is very long for this instance size).

Reply:
If we set`nbIter=10000` will it be sufficient?

Reply:
Yes absolutely